domain/src/main/java/life/qbic/domain/usermanagement/registration/Registration.java

```diff
@@ -69,28 +74,35 @@ public void setRegisterUserOutput(RegisterUserOutput registerUserOutput) {
   @Override
   public void register(String fullName, String email, char[] rawPassword) {
     try {
-      userRegistrationService.registerUser(fullName, email, rawPassword);
-      registerUserOutput.onSuccess();
-    } catch (UserException e) {
-      if (containsSuppressedUserExceptions(e)) {
-        listUserExceptionReasons(e).forEach(reason -> registerUserOutput.onFailure(reason));
-      } else {
-        registerUserOutput.onFailure(e.getReason());
+      RegistrationResponse registrationResponse = userRegistrationService.registerUser(fullName,
+          email, rawPassword);
+      if (registrationResponse.hasFailures()) {
+        registerUserOutput.onUnexpectedFailure(build(registrationResponse));
+        return;
       }
+      registerUserOutput.onUserRegistrationSucceeded();
     } catch (Exception e) {
-      registerUserOutput.onFailure("Unexpected error occurred.");
+      registerUserOutput.onUnexpectedFailure("Unexpected error occurred.");
     }
   }
 
-  private boolean containsSuppressedUserExceptions(UserException e) {
-    return e.getSuppressed().length > 0;
-  }
+  private UserRegistrationException build(RegistrationResponse registrationResponse) {
+    var builder = UserRegistrationException.builder();
 
-  private List<String> listUserExceptionReasons(UserException e) {
-    List<String> userExceptionReasons = new ArrayList<>();
-    Arrays.stream(e.getSuppressed()).toList()
-        .forEach(userException -> userExceptionReasons.add(userException.getMessage()));
-    return userExceptionReasons;
+    for (RuntimeException e : registrationResponse.failures()) {
+      if (e instanceof EmailValidationException) {
+        builder.withEmailFormatException((EmailValidationException) e);
+      }
+      if (e instanceof PasswordValidationException) {
+        builder.withInvalidPasswordException((PasswordValidationException) e);
+      }
+      if (e instanceof InvalidFullNameException) {
+        builder.withFullNameException((InvalidFullNameException) e);
+      } else {
+        builder.withUnexpectedException(e);
+      }
```

Comment:
what happens when more than 1 unexpected exception occurs?

Reply:
In theory if an unexpected Exception occurs the registration use case should terminate. Currently we don't check for multiple unexpected exceptions